### PR TITLE
W8 - D3 - BaoLe - Canvas

### PR DESCRIPTION
**Pull request by at-baole**
#49

### CODE REVIEW

File: app/src/at-baole/res/layout/activity_canvas.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (at-trungnguyen) [inline app/src/at-baole/res/layout/activity_canvas.xml]: no need `LinearLayout` here. you can define `<asiantech.internship.summer.canvas.VerticalChartView` as parent.

File: app/src/at-baole/java/asiantech/internship/summer/canvas/VerticalChartView.java [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+        }
+    }
+
+    private void drawChartColumn(Canvas canvas, int width, int height) {
+        int smallColumnDistance = 25;
+        int largeColumnDistance;
+        int startColumnPosition = width / 8;
+        int startColumnHeight = height * 3 / 4;
+        int columnUnit = height / 14;
+        int moneyUnit = 20000;
```
Comment (at-trungnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/canvas/VerticalChartView.java]: why `2000` it depend on value you input.

File: app/src/at-baole/java/asiantech/internship/summer/canvas/VerticalChartView.java [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        int startWidthPosition = 50;
+        int firstNumberHeight = height * 3 / 4;
+
+        for (int i = 0; i <= 7; i++) {
+            canvas.drawText("$" + i * moneyUnit, startWidthPosition, firstNumberHeight - i * height / 14 + 15, mPaint);
+        }
+    }
+
+    private void addData() {
+        mMoneyLists.add(new Money(getResources().getString(R.string.abbrJanuary), 70000, 10000));
```
Comment (at-trungnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/canvas/VerticalChartView.java]: try with larger value `700000`?

File: app/src/at-baole/java/asiantech/internship/summer/canvas/VerticalChartView.java [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+        while (twoFirstDigits >= 100) {
+            twoFirstDigits = twoFirstDigits / 10;
+            countDigits++;
+        }
+
+        int quotient = (int) twoFirstDigits / 7;
+        int surplus = (int) twoFirstDigits % 7;
+        if (surplus == 0) {
+            return quotient * Math.pow(10, countDigits);
+        } else return (quotient + 1) * Math.pow(10, countDigits);
```
Comment (at-dev-support-01) [inline app/src/at-baole/java/asiantech/internship/summer/canvas/VerticalChartView.java]: WARNING: [IfElseStmtsMustUseBraces] Avoid using if...else statements without curly braces
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/braces.html#IfElseStmtsMustUseBraces

**Comment by at-baole** [comment]
Thanks for reviewing. I hope my modifications meet your expectations.

File: app/src/at-baole/java/asiantech/internship/summer/canvas/VerticalChartView.java [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+
+    private void drawChartColumn(Canvas canvas, int width, int height) {
+        int smallColumnDistance = 25;
+        int largeColumnDistance;
+        int startColumnPosition = width / 8;
+        int startColumnHeight = height * 3 / 4;
+        int columnUnit = height / 14;
+        mPaintText.setTextSize(getResources().getDimension(R.dimen.smallTextSize));
+
+        for (int i = 0; i < mMoneyLists.size(); i++) {
```
Comment (at-trungnguyen) [inline app/src/at-baole/java/asiantech/internship/summer/canvas/VerticalChartView.java]:
```
int size = mMoneyLists.size();
for (int i = 0; i < size; i++) {
```

**Comment by at-trungnguyen** [comment]
@at-baole Thanks, LGTM!

**Pull request merged**